examples/prompting/Chain_of_thought_prompting.ipynb

```diff
@@ -0,0 +1,223 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Ne-3gnXqR0hI"
+      },
+      "outputs": [],
+      "source": [
+        "!pip install -U -q google-generativeai"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "EconMHePQHGw"
+      },
+      "outputs": [],
+      "source": [
+        "import google.generativeai as genai\n",
+        "\n",
+        "from IPython.display import Markdown"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eomJzCa6lb90"
+      },
+      "source": [
+        "## Configure your API key\n",
+        "\n",
+        "To run the following cell, your API key must be stored it in a Colab Secret named `GOOGLE_API_KEY`. If you don't already have an API key, or you're not sure how to create a Colab Secret, see [Authentication](https://github.com/google-gemini/cookbook/blob/main/quickstarts/Authentication.ipynb) for an example."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "v-JZzORUpVR2"
+      },
+      "outputs": [],
+      "source": [
+        "from google.colab import userdata\n",
+        "GOOGLE_API_KEY=userdata.get('GOOGLE_API_KEY')\n",
+        "\n",
+        "genai.configure(api_key=GOOGLE_API_KEY)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LlRYq9JSeRzR"
+      },
+      "source": [
+        "## Example\n",
+        "\n",
+        "Here is an example with chain of thought prompting. Take a look at what the result would be without chain of thought implemented."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "8oS9LnnXXedG"
+      },
+      "outputs": [],
+      "source": [
+        "model = genai.GenerativeModel(model_name='gemini-1.5-flash-latest', generation_config={\"temperature\": 0})"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0u1IjBOmeQgG"
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/markdown": "Here's how to solve this problem:\n\n**Understanding the Relationship**\n\n* **People and Donuts:** If 5 people make 5 donuts, that means each person makes 1 donut in 5 minutes.\n* **Time:**  The time it takes to make a donut doesn't change with more people.\n\n**Solving the Problem**\n\n1. **Individual Rate:** Each person makes 1 donut every 5 minutes.\n2. **Total Donuts:** With 25 people, they can make 25 donuts every 5 minutes (25 people * 1 donut/person = 25 donuts).\n3. **Time for 100 Donuts:** To make 100 donuts, it would take 20 minutes (100 donuts / 25 donuts/5 minutes = 20 minutes).\n\n**Answer:** It would take 25 people **20 minutes** to make 100 donuts. \n",
```

Comment:
This already looks like chain of thought. It's like it's almost the default behavior. We might need to update the text to reflect that.

Here you almost want to say "don't reply anything except the answer" to show it performing poorly.